### PR TITLE
Add "happy path" tests for all backend routes

### LINKED ISSUE #75: Add tests for basic business logic on server
We should probably have some minimal testing on the server side. As app grows and more folks are contributing, risk of regression increases.

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Closes #75 

This PR brings setup for tests and a single "happy path" test for each of the current routes on the server side (except for the static files route).

This PR also brings a series of convenience functions that are meant to be used to standardized how our API represents contributors, contributions, and candidates. These are `apiReprCandidates`, `apiReprContributions` and `apiReprContributors` at the top of `/server/api.js`. This effectively reduces the number of key-value pairs we send back to represent these resources. 

## To run tests

This PR adds new dev dependencies, so before running tests, be sure to run `npm install` from the server folder.

Also, copy over the value for `TEST_DATABASE_URL` from `sample.env`.

To run tests, from the server folder, run `npm test`.

There are presently 9 tests, all of which should pass. After the tests run, note that it takes up to ~10 seconds for the routine to finish, but it *should* in fact finish, and we shouldn't need to resort to using the `--exit` flag with our mocha command.

### CODE REVIEW

**Comment by SchaeStewart** [comment]
This looks pretty solid over all. Regarding the test db, would it make more sense if we just created another DB in the docker-compose.yml just for testing?  
We could have its credentials match the default `TEST_DATABASE_URL`, so then new contributors can still copy the `sample.env` to `.env` and everything just works.

**Comment by benjaminEwhite** [comment]
Excellent idea @SchaeStewart re: additional db  in docker compose.

File: server/api.js [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+    const decodedName = decodeURIComponent(name)
+
+    const contributors = await searchContributors(
+      decodedName,
+      offset,
+      limit,
+      TRIGRAM_LIMIT
+    )
+    return res.send({
+      data: contributors.data.map((contributor) =>
```
Comment (SchaeStewart) [inline server/api.js:106]: map statements can be shortened to `contributors.data.map(apiReprContributor)`. I don't care if we change them, up to you

File: server/db/index.js [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
         }
       : undefined,
 })
+
 const query = (text, params) => pool.query(text, params)
 /**
  * @returns {Promise<pg.PoolClient>}
  */
-const getClient = () => pool.connect()
+const getClient = async () => await pool.connect()
```
Comment (SchaeStewart) [inline server/db/index.js:22]: Why is this `async/await`?  `pool.connect()` already returns a promise

Comment (benjaminEwhite) [reply server/db/index.js:22]: I'll try reverting and see if tests hang. When i was trying to track down whatever was causing tests to hang, I made this change, which was a bit of "cargo cult" programming tbh.

**Pull request merged**